standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5882,7 +5884,7 @@ public String getVersion() throws TException {
   @Override
   public void alter_table(final String dbname, final String name,
                           final Table newTable)
-      throws InvalidOperationException, MetaException {
+      throws InvalidOperationException, MetaException, NoSuchObjectException {
```

Comment:
so I am assuming the get_database_core() call now throws a NoSuchObjectException that we now have to account for.
I realize the current code is a bit inconsistent when it comes to exception handling. so for drop_table(), if the table does not exist, we throw a NoSuchObjectException. But for alter_table() if the table does not exist, we catch the NoSuchObjectException and rethrow it as an InvalidOperationException. 
Ideally they should be consistent.

But I am a bit concerned about throwing a new exception from HMS. It might be backwards-incompatible with the current code. Could you instead catch this and re-throw as InvalidOperationException as well.